server/plugin/api.go

```diff
@@ -275,12 +276,20 @@ func (p *Plugin) handleCreateSubscription(w http.ResponseWriter, r *http.Request
 func (p *Plugin) handleGetSubscriptions(w http.ResponseWriter, r *http.Request) {
 	mattermostUserID := r.Header.Get(constants.HeaderMattermostUserID)
 	subscriptionList, err := p.Store.GetAllSubscriptions(mattermostUserID)
+
 	if err != nil {
 		p.API.LogError(constants.FetchSubscriptionListError, "Error", err.Error())
 		p.handleError(w, r, &serializers.Error{Code: http.StatusInternalServerError, Message: err.Error()})
 		return
 	}
 
+	offset, limit, err := p.GetOffsetAndLimitFromQueryParams(r)
+	if err != nil {
+		p.API.LogError(constants.InvalidPaginationQueryParam, "Error", err.Error())
+		p.handleError(w, r, &serializers.Error{Code: http.StatusBadRequest, Message: constants.InvalidPaginationQueryParam})
```

Comment:
I think you should just log the error in server logs and shouldn't return any error in the API response. If the query params for pagination are wrong, you can just use the default values for those and log the error in server logs.